code/02_create-figures.R

```diff
@@ -0,0 +1,310 @@
+#
+# Written/run on - RStudio Server
+# Version of R - 3.6.1
+#
+# Description - Create and save figures for summary and report.
+#
+# Approximate run time - xx minutes
+#########################################################################
+
+
+### 1 - Setup environment and load functions ----
+
+source(here::here("code", "00_setup-environment.R"))
+source(here::here("functions", "summarise_data.R"))
+
+
+### 2 - Read in basefile ----
+
+basefile <- read_rds(here("data", "basefiles", 
+                          glue("{pub_date}_base-file.rds")))
+
+
+### 3 - Figure 1 - Trend Bar Chart ----
+
+fig1 <- 
+  
+  basefile %>%
+  summarise_data(include_years = "all",
+                 format_numbers = FALSE) %>%
+  pivot_longer(cols = qom:qom_hosp,
+               names_to = "qom") %>%
+  mutate(qom = if_else(qom == "qom_hosp",
+                       "Hospital",
+                       "Home/Community")) %>%
+  
+  # Control order of stacks in chart
+  mutate(qom = fct_relevel(as.factor(qom), "Hospital")) %>%
+  
+  ggplot(aes(x = fy, y = value, fill = qom)) +
+  geom_bar(position = "stack", stat = "identity", width = 0.5, show.legend = T) +
+  scale_fill_manual(values = c("#00a2e5", "#004785")) +
+  theme(panel.background = element_blank(),
+        panel.grid.major.x = element_blank(),
+        panel.grid.major.y = element_blank(),
+        axis.title.x = element_text(size = 10, face = "bold"),
+        axis.title.y = element_text(size = 10, face = "bold"),
+        axis.text = element_text(size = 10),
+        axis.text.x = element_text(angle = 45),
+        legend.title = element_blank()) +
+  xlab("Financial Year of Death") + 
+  ylab("Percentage")
+
+ggsave(here("markdown", "figures", "figure-1.png"), 
+       plot = fig1,
+       width = 17.49, height = 9.03, 
+       units = "cm", device = "png", dpi = 600)
+
+ggsave(here("markdown", "figures", "figure-1-summary.png"), 
+       plot = fig1,
+       width = 17, height = 7, 
+       units = "cm", device = "png", dpi = 600)
+
+
+### 4 - Figure 2 - Health Board Map ----
+
+fig2 <- shapefile()
+
+fig2@data %<>%
+  left_join(summarise_data(basefile, hb,
+                           format_numbers = FALSE) %>% 
+              mutate(hb = substring(hb, 5)) %>%
+              select(hb, qom),
+            by = c("HBName" = "hb"))
+
+fig2@data$id <- rownames(fig2@data)
+
+fig2 <-
+  full_join(tidy(fig2, region = "id"),
+            fig2@data,
+            by = "id")
+
+fig2 <-
+  
+  ggplot() +
+  geom_polygon(data = fig2,
+               aes(x = long,
+                   y = lat,
+                   group = group,
+                   fill = qom),
+               colour = "white",
+               size = 0.3) +
+  scale_fill_continuous(trans = "reverse") +
+  theme(panel.background = element_blank(),
+        panel.grid = element_blank(),
+        axis.text = element_blank(),
+        axis.title = element_blank(),
+        axis.ticks = element_blank(),
+        legend.title = element_blank())
```

Comment:
For the annotation, I did some tests and think using the following code will look good for this map. You can add it after theme():

```
annotate("text", x = 4e+05, y = 620000, label = paste0("NHS Borders: ", min(round_half_up(fig2$qom, 1)), "%"), size = 3.5) +
annotate("text", x = 4e+05, y = 1100000, label = paste0("NHS Shetland: ", max(round_half_up(fig2$qom, 1)), "%"), size = 3.5)
```
The only problem for the future will be the manual change for the labels' position (x and y figures) if the health boards for minimum and maximum change.

Reply:
Thanks Tina - I think this is a good solution for now. I'll add an issue so we can look into how to automate this at a later date.